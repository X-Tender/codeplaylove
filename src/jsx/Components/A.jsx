import React from 'react';
import PropTypes from 'prop-types';

const A = ({ href, target, children, title }) => {
	return (
		<a
			href={href}
			target={target}
			rel={target === '_blank' ? 'noopener noreferrer' : null}
			title={title || children}
		>
			{children}
		</a>
	);
};

A.propTypes = {
	href: PropTypes.string.isRequired,
	target: PropTypes.string,
	title: PropTypes.string,
	children: PropTypes.string.isRequired,
};

A.defaultProps = {
	target: '_SELF',
	title: null,
};

export default A;
