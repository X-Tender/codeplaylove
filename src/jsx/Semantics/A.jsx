import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
	href: PropTypes.string.isRequired,
	target: PropTypes.string,
	title: PropTypes.string,
	children: PropTypes.node.isRequired,
};

const defaultProps = {
	target: '_SELF',
	title: null,
};

const A = ({ href, target, children, title }) => (
	<a
		href={href}
		rel={target === '_blank' ? 'noopener noreferrer' : null}
		target={target}
		title={title || children}
	>
		{children}
	</a>
);

A.propTypes = propTypes;
A.defaultProps = defaultProps;

export default A;
