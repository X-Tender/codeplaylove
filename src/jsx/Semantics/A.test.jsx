import React from 'react';
import { shallow } from 'enzyme';
import A from './A';

describe('A', () => {
	const href = 'https://codeplaylove.de';
	const target = '_blank';
	const title = 'CodePlayLove';
	const children = 'My portfolio';

	describe('with all props passed', () => {
		const AWithAllProps = shallow(
			<A href={href} target={target} title={title}>
				{children}
			</A>
		);

		it('renders correctly', () => {
			expect(AWithAllProps).toMatchSnapshot();
		});

		it('renders the correct props', () => {
			const a = AWithAllProps.find('a');
			expect(a.prop('href')).toEqual(href);
			expect(a.prop('target')).toEqual(target);
			expect(a.prop('title')).toEqual(title);
			expect(a.prop('rel')).toEqual('noopener noreferrer');
			expect(a.text()).toEqual(children);
		});
	});

	describe('with required props passed', () => {
		const AWithReqProps = shallow(<A href={href}>{children}</A>);

		it('renders correctly', () => {
			expect(AWithReqProps).toMatchSnapshot();
		});

		it('renders the correct props', () => {
			const a = AWithReqProps.find('a');
			expect(a.prop('href')).toEqual(href);
			expect(a.prop('target')).toEqual('_SELF');
			expect(a.prop('title')).toEqual(children);
			expect(a.prop('rel')).toEqual(null);
			expect(a.text()).toEqual(children);
		});
	});
});
