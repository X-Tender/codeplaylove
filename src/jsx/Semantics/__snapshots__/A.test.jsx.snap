// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A with all props passed renders correctly 1`] = `
<a
  href="https://codeplaylove.de"
  rel="noopener noreferrer"
  target="_blank"
  title="CodePlayLove"
>
  My portfolio
</a>
`;

exports[`A with required props passed renders correctly 1`] = `
<a
  href="https://codeplaylove.de"
  rel={null}
  target="_SELF"
  title="My portfolio"
>
  My portfolio
</a>
`;
